libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -213,9 +270,19 @@ void AP_Frsky_Telem::send_SPort_Passthrough(void)
     for (int16_t i = 0; i < numc; i++) {
         prev_byte = _passthrough.new_byte;
         _passthrough.new_byte = _port->read();
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+        if (parse_sport_telemetry_data(_passthrough.new_byte, _SPort_bidir.parse_state)) {
```

Comment:
functions + state == object.  Why are we passing state through as a separate thing here rather than having an instance of a class to hold the state etc?

Reply:
because the parser is in frsky.h and is not in a class. My intention here is to allow the reuse of the parser and all frsky.h defs. One such use is the frsky sensors PR https://github.com/ArduPilot/ardupilot/pull/14583

Reply:
full refactoring in upcoming commits